### PR TITLE
add clean-clear-sky and clean-sky radiation calls

### PR DESCRIPTION
**Pull request by mahf708**
Adds two additional radiation diagnostic calls following the logic of clear-sky calls. These are clean-clear-sky (neither clouds nor aerosols) and clean-sky (no aerosols). The latter in enabled by instantiating a second optics and never endowing it with aerosols, while the former is called before the other optics are endowed with aerosols and clouds.

TODO:
- [x] some tests
- [x] performance guard
- [x] add docs

### CODE REVIEW

**Comment by jeff-cohere** [comment]
Looks good to me so far!

**Review by bartgol — DISMISSED** [review]

File: components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp
Comment (brhillman) [inline components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp:769]: This is a fairly expensive operation, and will be identical to the operation with `optics` with aerosol. I think it might be preferable to simple do a deep copy of the `optics` computed, unless this is meant to allow for changing gas optics as well (but we'd probably want to implement that a little differently to mimic the EAM-style diag calls).

File: components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp
Comment (brhillman) [inline components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp:904]: Again, I think this is fairly expensive, and we might think about just doing a deep copy of the `optics` object.

File: components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp
Comment (brhillman) [inline components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp:911]: Isn't this one clean-clear-sky fluxes?

**Comment by brhillman** [comment]
Minor comments. This PR will have an impact on radiation performance. This is probably fairly minimal relative to total throughput since we do not run radiation every timestep, but we should evaluate the impact. Note that the additional calls to gas_optics are unnecessary, and we could probably alternatively just deep copy the optics object after the first call to gas optics. This would recover some performance, and if I remember right the gas_optics call is not exactly cheap. We might also consider NOT doing these extra calls if we are not outputting the additional diag calls. That's the strength of the EAM approach that does these in a loop, which we may consider in the future, but this is certainly the simplest approach for getting this functionality in ASAP.

**Comment by PeterCaldwell** [comment]
@mahf708 - thanks for these heuristic results. I'd also be curious how much the total wall clock time for the simulations increases from double-call radiation.

File: components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp
Comment (bartgol) [reply components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp:824]: I'm not familiar with the `OpticalProps2str` data structure, so I can't comment. I'll let Ben decide.

Comment (brhillman) [reply components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp:824]: My initial reaction to this is that I don't personally like the idea of adding a decrement method to the external optics class, but I'll have to think about that a little more. I think it makes more sense to create the combinations we want here by addition rather than subtraction, and in the future based on a set of lists of constituents for each diag call.

File: components/eamxx/cime_config/namelist_defaults_scream.xml
Comment (bartgol) [inline components/eamxx/cime_config/namelist_defaults_scream.xml:311]: Would you mind adding the `type="logical"` specifier here, and perhaps, also a `doc="blah"` string? It helps with the auto-generated documentation.

Comment (bartgol) [reply components/eamxx/cime_config/namelist_defaults_scream.xml:311]: If you don't mind, and you have time to spare (haha), you could also add it for some of the other vars here (not the ones inherited from `atm_proc_base`, only the rrtmgp-specific ones).

File: components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp
Comment (bartgol) [inline components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp:124]: Don't change it, but FYI, if the type can be inferred from the default, you don't need the template. In practice, the _only_ case where you need to specify the type is when you pass a C-string as default, since we store strings as C++ strings. So this
```C++
params.get("my_string","my_value");
```
will fail at compile time (there's some overload that is not specified for C-strings), and you need to instead do
```C++
params.get<std::string>("my_string","my_value");
```
For virtually any other type we use, the provided default value is enough.

File: components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp
Comment (bartgol) [reply components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp:824]: I'm not really following. If/when you want to implement this, you can of course ping me and go over the details, if you need.

**Review by bartgol — DISMISSED** [review]

**Comment by jeff-cohere** [comment]
This PR has accumulated some history and, while it's not huge, has enough detail in it that I suspect I'm not following it as closely as I need to in order to understand the details. I didn't see anything that concerns me that hasn't been pointed out by others. @mahf708 , let me know if there's anything in particular you'd like me to examine more closely.

File: components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp
Comment (brhillman) [inline components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp:225]: This is probably not worth changing (especially since I want for us to refactor this soon to generalize to N-diag calls), but I think it might have been better to order the args to these calls as allsky, clrsky, clnsky, clnclrsky just to keep the "always on" fields first in the list (the cln versions are only computed if the extra bool args are true).

File: components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp
Comment (brhillman) [reply components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp:824]: Naser and I talked about this a week or two ago, I actually have a slightly different idea about how to handle this.

File: components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp
Comment (brhillman) [inline components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp:823]: I would move this to the extra_clnsky_diag logical chunk below to keep everything related to the extra_cldsky_diag call together so it's more obvious what goes into that calculation.

File: components/eamxx/src/physics/rrtmgp/tests/rrtmgp_tests.cpp
Comment (brhillman) [inline components/eamxx/src/physics/rrtmgp/tests/rrtmgp_tests.cpp:249]: Should this have any additional text string other than just the integer number with no context?

File: components/eamxx/src/physics/rrtmgp/tests/rrtmgp_unit_tests.cpp
Comment (brhillman) [inline components/eamxx/src/physics/rrtmgp/tests/rrtmgp_unit_tests.cpp:846]: I would pull this block out of the PR. If it isn't ready to include, let's not commit commented out code.

File: components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp
Comment (brhillman) [reply components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp:794]: I prefer simpler and clearer to fine-grain control, personally.

File: components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp
Comment (brhillman) [reply components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp:964]: Again, I prefer simple/clear to fine-grained control (too many options and duplicate code blocks clutters an interface unnecessarily unless there's good reason for it). But, that shouldn't hold up the PR.

**Review by brhillman — APPROVED** [review]

**Pull request merged**
